backend/utils/database.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any
+
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql as sa_pg
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import ColumnElement, func
+
+
+def CustomJSON(**kwargs: Any) -> sa.JSON:
+    """Custom SQLAlchemy JSON type that uses JSONB on PostgreSQL."""
+    return sa.JSON(**kwargs).with_variant(sa_pg.JSONB(**kwargs), "postgresql")
+
+
+def is_postgresql(conn: sa.Connection) -> bool:
+    return conn.engine.name == "postgresql"
+
+
+def json_array_contains_value(
```

Comment:
surprised sqlalchemy doesn't offer a utility for this

Reply:
I haven't found a way for MySQL to provide a similar functionality to PostgreSQL's `?` operator, which checks if a value exists in a JSONB array, or as a key for a JSONB object.

Maybe that's why it's not simple to provide this in SQLAlchemy, unless without knowing which kind of value the JSON column will contain:

```sql
SELECT JSON_CONTAINS('{"a": 1, "b": 2, "c": {"d": 4}}', '"a"', '$');  -- FALSE
SELECT JSON_CONTAINS('["a", "b", "c"]', '"a"', '$');  -- TRUE

SELECT JSON_CONTAINS_PATH('{"a": 1, "b": 2, "c": {"d": 4}}', 'one', '$.a');  -- TRUE
SELECT JSON_CONTAINS_PATH('["a", "b", "c"]', 'one', '$.a');  -- FALSE
```

If someone finds a way to do that, I think that could be contributed to SQLAlchemy, as a `Comparator.has_key` function for the `mysql.JSON` dialect type, similar to [the PostgreSQL implementation](https://github.com/sqlalchemy/sqlalchemy/blob/c6be0c1f205cf83467d68eb40fb650b9274519f9/lib/sqlalchemy/dialects/postgresql/json.py#L264-L268).